lib/autoupdate/agent/setup.go

```diff
@@ -211,16 +216,34 @@ func (ns *Namespace) Setup(ctx context.Context) error {
 	if err != nil {
 		return trace.Wrap(err, "failed to write teleport-update systemd config files")
 	}
-	svc := &SystemdService{
+	timer := &SystemdService{
 		ServiceName: filepath.Base(ns.updaterTimerFile),
 		Log:         ns.log,
 	}
-	if err := svc.Sync(ctx); err != nil {
+	if err := timer.Sync(ctx); err != nil {
 		return trace.Wrap(err, "failed to sync systemd config")
 	}
-	if err := svc.Enable(ctx, true); err != nil {
+	if err := timer.Enable(ctx, true); err != nil {
 		return trace.Wrap(err, "failed to enable teleport-update systemd timer")
 	}
+	if ns.name == "" {
+		oldTimer := &SystemdService{
+			ServiceName: deprecatedTimerName,
+			Log:         ns.log,
+		}
+		// If the old teleport-upgrade script is detected, disable it to ensure they do not interfere.
+		// Note that the schedule is also set to nop by the Teleport agent -- this just prevents restarts.
+		enabled, err := oldTimer.IsEnabled(ctx)
+		if err != nil {
+			return trace.Wrap(err, "failed to determine if deprecated teleport-upgrade systemd timer is enabled")
+		}
+		if enabled {
+			if err := oldTimer.Disable(ctx, true); err != nil {
+				return trace.Wrap(err, "failed to disable deprecated teleport-upgrade systemd timer")
```

Comment:
Same, should we log but keep going?

Reply:
On the fence about this. If `teleport-update enable` succeeds, but leaves the deprecated updater running, the agent will restart at arbitrary times. That state isn't fatal, but could be very disruptive. That said, the most likely reason for this to fail would be if someone removed the deprecated timer between `IsEnabled` and `Disable`, which is okay.

I'll switch it to a warning.